content/en/blog/2025-12-12-flux-aio-kubernetes-mtls-and-the-chicken-and-egg-problem/index.md

```
@@ -0,0 +1,132 @@
+
+But how do we organize access for Flux to the internal HTTP server running as a pod within the same cluster?
+
+Obviously, without CNI and kube-proxy, it won't be able to reach this pod by its persistent name (CoreDNS also depends on CNI and kube-proxy).
+
+There were several options, such as adding our HTTP server as a sidecar to Flux, or pinning it via nodeAffinity to the same node and forcing Flux to access it on localhost. But [@lllamnyp](https://github.com/lllamnyp) proposed a more elegant solution—routing Flux through the Kubernetes API.
+
+The idea immediately seemed good to me, as it also addresses the need for an open port on the node (though it later turned out this wasn't the case).
+
+Thus, we run a pod `cozystack-assets-0` and can access its contents via:
+
+```
```

markdownlint-cli2 (0.18.1)

Fenced code blocks should have a language specified

(MD040, fenced-code-language)